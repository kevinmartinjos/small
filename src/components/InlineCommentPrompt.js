import React from 'react';
import {Button, Glyphicon} from 'react-bootstrap';
import InlineCommentInputContainer from './InlineCommentInputContainer';
import '../styles/InlineCommentPrompt.css';

var PropTypes = React.PropTypes;

var InlineCommentPrompt = React.createClass({
	propTypes: {
		x: PropTypes.number.isRequired,
		y: PropTypes.number.isRequired,
		domId: PropTypes.string,
		submitHandlerCallback: PropTypes.func.isRequired,
		cancelCallback: PropTypes.func.isRequired
	},
	getInitialState(){
		return {
			style:{
				position: 'absolute',
				left: this.props.x,
				top: this.props.y,
			},
			prompt: true
		};
	},
	handlePromptClick(){
		this.setState({
			prompt: false
		});
	},
	storeCommentInputContainer(ref){
		this.commentInputContainer = ref;
	},
	render() {
		return(
			<div className="InlineCommentPrompt" ref="child" style={this.state.style}>
				{this.state.prompt ?
					(<Button className='makeComment' bsSize='lg' >
						<Glyphicon glyph="pencil" onClick={this.handlePromptClick}/>
					</Button>)
					:
					(
						<InlineCommentInputContainer
						domId={this.props.domId}
						submitHandlerCallback={this.props.submitHandlerCallback}
						cancelCallback={this.props.cancelCallback}
						ref={this.storeCommentInputContainer}
						/>)
				}
			</div>
		);
	}
});

export default InlineCommentPrompt;
