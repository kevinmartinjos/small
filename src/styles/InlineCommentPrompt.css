.InlineCommentPrompt{
	z-index: 1000;
}

.InlineCommentPrompt .makeComment{
	font-size: 32px;
	padding-left: 8px;
	padding-right: 8px;
	padding-top: 2px;
	padding-bottom: 2px;
	color: #037F8C;
	border: 1px solid #037F8C;
}