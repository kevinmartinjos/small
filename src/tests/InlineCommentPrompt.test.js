import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import InlineCommentPrompt from '../components/InlineCommentPrompt';
import ReactTestUtils from 'react-addons-test-utils';


test('InlineCommentPrompt should be rendered', () => {
	function submitHandlerCallback(){};
	function cancelCallback(){};
	const component = renderer.create(
		<InlineCommentPrompt x={1} y={1} 
			submitHandlerCallback={submitHandlerCallback}
			cancelCallback={cancelCallback}
		/>
	);
	let tree = component.toJSON();
  	expect(tree).toMatchSnapshot();
});

test('Display InlineCommentInputContainer when pressed', () => {

	let component = ReactTestUtils.renderIntoDocument(<InlineCommentPrompt x={1} y={1} 
														submitHandlerCallback={submitHandlerCallback}
														cancelCallback={cancelCallback}
													/>);
	component.handlePromptClick();
	let node = ReactTestUtils.findRenderedDOMComponentWithClass(component, 'InlineCommentInputContainer');
	expect(node).not.toBe(undefined);
});
