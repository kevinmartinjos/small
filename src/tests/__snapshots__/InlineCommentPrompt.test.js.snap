exports[`test InlineComment should be rendered 1`] = `
<div
  className="InlineCommentPrompt"
  style={
    Object {
      "left": 1,
      "position": "absolute",
      "top": 1,
    }
  }>
  <button
    className="btn btn-sm btn-default"
    disabled={false}
    onClick={[Function]}
    type="button">
    c
  </button>
</div>
`;
